src/scripts/create_hunter_csv.py

```diff
@@ -0,0 +1,477 @@
+
+    Args:
+        json_paths: List[str]
+                    A list of json paths with performance results and related metrics.
+        is_prospective: bool
+                    Whether the analysis is prospective (True by default).
+
+    Returns:
+            A dataframe of test type-specific performance results.
+    """
+
+    if len(json_paths) == 0:
+        logging.error(
+            "The 'json_paths' is empty; "
+            "thus, an empty dataframe is being returned."
+        )
+        return pd.DataFrame()
+
+    # Access performance-report.json
+    with open(json_paths[0]) as json_file:
+        data = json.load(json_file)
+        if len(data['stats']) == 0:
+            logging.error(
+                "The 'stats' list in the 'data' dictionary is empty; "
+                "please ensure that the test has output the statistics into it."
+            )
+            return pd.DataFrame()
+
+        # Get only read/write-result-success dictionaries.
+        read_dict = get_relevant_dict(data, 'read')
+        write_dict = get_relevant_dict(data, 'write')
+        # Get dataframe with relevant read/write column names, rename columns to shorten their names.
+        raw_hunter_metrics_df = extract_metrics_df(read_dict, write_dict)
+        raw_hunter_metrics_df['opRate'] = raw_hunter_metrics_df['opRate'].str.rstrip(
+            ' op/sec')
+        for col_name in raw_hunter_metrics_df.columns:
+            if col_name not in ['totalOps', 'opRate']:
+                raw_hunter_metrics_df[col_name] = raw_hunter_metrics_df[col_name].str.rstrip(
+                    ' ms')
+        # Get date based on path and add time for compatibility with hunter.
+        date_val = json_paths[0].split(os.sep)[4]
+        # As at 11pm UTC
+        date_val_w_time = f"{date_val.replace('_', '-')}{' '}{LWT_TEST_RUN_EXEC_TIME}{' +0000'}"
+
+        # Prospective case is when running hunter nightly; retrospective is when running it
+        # on already output dates-related folders of performance results (one-off analysis).
+        if is_prospective:
+            cassandra_git_short_hash, fallout_tests_git_short_hash = \
+                get_commit_hash_cass_fall_tests(is_prospective=is_prospective)
+
+        else:
+            # Swap month and day to match expected format for the git log command below
+            list_of_elems = date_val.split('_')
+            # Plus one day to then get the date's Git sha (as using 'until' in git log command)
+            date_sorted = datetime(
+                int(list_of_elems[0]), int(
+                    list_of_elems[1]), int(list_of_elems[2])
+            ) + timedelta(days=1)
+            date_sorted_y_d_m = date_sorted.strftime(FMT_Y_D_M)
+            cassandra_git_short_hash, fallout_tests_git_short_hash = \
+                get_commit_hash_cass_fall_tests(
+                    date_sorted_y_d_m, is_prospective)
+
+        # cd back into hunter_csv
+        cd_into_proj_dir()
```

Comment:
Can we make this stateless? e.g. return the full path the project directory as a string instead of altering the current working directory.